float64 mass 
---
bool success